### PR TITLE
Even less integral domain

### PR DESCRIPTION
**Pull request by fchapoton**
remove two uses of the auld class `IntegralDomain`

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.

### CODE REVIEW

**Comment by fchapoton** [comment]
somehow this is breaking the `_singular_init_` of polynomials

File: src/sage/rings/polynomial/polynomial_ring.py [hunk 1]
```diff
@@ -149,7 +149,7 @@
 from sage.categories.principal_ideal_domains import PrincipalIdealDomains
 from sage.categories.rings import Rings
 
-from sage.rings.ring import (Ring, IntegralDomain)
+from sage.rings.ring import Ring
```
Comment (mantepse) [inline src/sage/rings/polynomial/polynomial_ring.py]:
```suggestion
from sage.rings.ring import Ring, CommutativeRing
```

File: src/sage/rings/polynomial/polynomial_ring.py [hunk 2]
```diff
@@ -1923,8 +1923,7 @@ def _roots_univariate_polynomial(self, p, ring=None, multiplicities=True, algori
         return roots
 
 
-class PolynomialRing_integral_domain(PolynomialRing_commutative, PolynomialRing_singular_repr,
-                                     IntegralDomain):
+class PolynomialRing_integral_domain(PolynomialRing_commutative, PolynomialRing_singular_repr):
```
Comment (mantepse) [inline src/sage/rings/polynomial/polynomial_ring.py]:
```suggestion
class PolynomialRing_integral_domain(PolynomialRing_commutative, PolynomialRing_singular_repr, CommutativeRing):
```

File: src/sage/rings/polynomial/multi_polynomial_ring.py [hunk 3]
```diff
@@ -936,15 +936,11 @@ def sum(self, terms):
         return MPolynomial_polydict(self, elt)
 
 
-class MPolynomialRing_polydict_domain(IntegralDomain,
-                                      MPolynomialRing_polydict):
+class MPolynomialRing_polydict_domain(MPolynomialRing_polydict):
```
Comment (mantepse) [inline src/sage/rings/polynomial/multi_polynomial_ring.py:937]: There is still an `import IntegralDomain` in line 63 or so.

File: src/sage/rings/polynomial/polynomial_ring.py (hunk 2 above)
Comment (mantepse) [reply src/sage/rings/polynomial/polynomial_ring.py]: Actually, the `CommutativeRing` might go into `PolynomialRing_commutative`, no?

**Review by mantepse — APPROVED** [review]
Works for me (I checked all the test failures, they are fake)

**Pull request merged**